src/game/moveFinding/findKnightBaseMoves.js

```diff
@@ -0,0 +1,29 @@
+import BoardCoordinates from "../board/boardCoordinates";
+class Knight {
+    
+    outOfBounds(i, j) {
+        return ( i < 0 || i >= 8 || j < 0 || j >= 8 );
+    }
+
+    findMovesx(boardSituation, startingCoordinates, pieceColor){
+        const knightMoves = [
+            [-1, -2], [-2, -1], [1, -2], [-2, 1],
+            [2, -1], [-1, 2], [2, 1], [1, 2]
+        ];
+        var moveX = startingCoordinates.x;
+        var moveY = startingCoordinates.y;
+        var nextMoves = [];
+            for(var move of knightMoves) {
+                var newMoveX = moveX + move[0];
+                var newMoveY = movey + move[1];
+                if( !outOfBounds(newMoveX, newMoveY)) {
+                            let possiblePosition = boardSituation[newMovex][newMovey];

```

Comment:
1. Sugerowałbym unikanie słówka "position" w nazwach zmiennych nie odnoszących się do pozycji tak, jak jest to rozumiane w naszym projekcie (jako reprezentacja rozstawienia wszystkich bierek na szachownicy).

2. **Argument boardSituation zawierać powinien obiekt klasy BoardSituation, który nie jest tablicą. Dostęp do tablicy reprezentującej rozstawienie figur na szachownicy uzyskac można przez boardSituation.position.**